### PR TITLE
1) implement Spring boot, added entities and connected with it app le…

### PR DESCRIPTION
**Pull request by AntonPanchishko**
…vels

### CODE REVIEW

File: src/main/java/com/example/bankservicedemo/service/impl/RoleServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    private final RoleRepository roleRepository;
+
+    @Override
+    public Role add(Role role) {
+        return roleRepository.save(role);
+    }
+
+    @Override
+    public Role getByName(String name) {
+        return roleRepository.getByRoleName(Role.RoleType.valueOf(name)).get();
```
Comment (vicire) [inline src/main/java/com/example/bankservicedemo/service/impl/RoleServiceImpl.java]: What happened if the name of the role doesn`t exist in DB?

File: src/main/java/com/example/bankservicedemo/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    public User update(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User get(Long id) {
+        return userRepository.findById(id).get();
```
Comment (vicire) [inline src/main/java/com/example/bankservicedemo/service/impl/UserServiceImpl.java]: Same here

File: src/main/java/com/example/bankservicedemo/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    public User getByPhone(String phone) {
+        return userRepository.getFirstByPhoneNumber(phone).get();
```
Comment (vicire) [inline src/main/java/com/example/bankservicedemo/service/impl/UserServiceImpl.java]: and here

**Review by vicire — CHANGES_REQUESTED** [review]

**Review by vicire — APPROVED** [review]

File: src/main/java/com/example/bankservicedemo/repository/UserRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.example.bankservicedemo.repository;
+
+import com.example.bankservicedemo.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> getFirstByPhoneNumber(String phoneNumber);
+
+    @Override
```
Comment (boroda4436) [inline src/main/java/com/example/bankservicedemo/repository/UserRepository.java]: Do we need `@Override` annotation here?

File: src/main/java/com/example/bankservicedemo/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    Optional<User> findById(Long id);
```
Comment (boroda4436) [inline src/main/java/com/example/bankservicedemo/repository/UserRepository.java]: Should we override the base method?

File: src/main/java/com/example/bankservicedemo/service/impl/RoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public Role add(Role role) {
+        return roleRepository.save(role);
+    }
+
+    @Override
+    public Role getByName(String name) {
+        return roleRepository.getByRoleName(Role.RoleType.valueOf(name))
+                .orElseThrow(() ->
+                        new NoSuchElementException("Can't find role with such name " + name));
```
Comment (boroda4436) [inline src/main/java/com/example/bankservicedemo/service/impl/RoleServiceImpl.java]: better throw your custom `NoSuchEntityException`

File: src/test/java/com/example/bankservicedemo/BankServiceDemoApplicationTests.java [hunk 7]
```diff
@@ -1,5 +1,6 @@
 package com.example.bankservicedemo;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (boroda4436) [inline src/test/java/com/example/bankservicedemo/BankServiceDemoApplicationTests.java]: Are you using this method?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**